src/storage-controller/src/rehydration.rs

```diff
@@ -433,10 +433,29 @@ where
             }
             StorageCommand::CreateSinks(exports) => {
                 for export in exports {
-                    self.sinks.insert(export.id, export.clone());
-                    // Initialize the uppers we are tracking
-                    self.uppers
-                        .insert(export.id, Antichain::from_elem(T::minimum()));
+                    let prev = self.sinks.insert(export.id, export.clone());
+                    assert!(
+                        prev.is_some() == export.update,
+                        "can only and must update source if RunSinkCommand is update"
```

Comment:
We should just make a sequence of `(GlobalId, Desc), (GlobalId, Desc)` to have upsert semantics without the update flag. The update flag can be synthetized by anyone observing the stream by checking if the id is already known

Reply:
Did this by removing the `update` field from both source and sink descriptors and only inserting e.g. tracked uppers when there is not some other value present.